2.1.cpp

```diff
@@ -0,0 +1,86 @@
+#define _USE_MATH_DEFINES 
+#include <iostream>
+#include <iomanip>
+#include <cmath>
+
+using namespace std;
+
+/**
+ * \brief Расчет периметра прямоугольного треугольника
+ * \param CatheterOne Длина первого катета
+ * \param CatheterTwo Длина второго катета
+ * \return Периметр прямоугольного треугольника
+ */
+double GetPerimeter(double CatheterOne, double CatheterTwo);
+
+/**
+ * \brief Расчет площади прямоугольного треугольника
+ * \param CatheterOne Длина первого катета
+ * \param CatheterTwo Длина второго катета
+ * \return Площадь прямоугольного треугольника
+ */
+double GetArea(double CatheterOne, double CatheterTwo);
+
+/**
+ * \brief Пользовательский выбор расчета  периметра (0) или площади (1)
+ */
+enum class Choice { perimeter, area };
+
+int main()
+{
+	// объявление переменных CatheterOne, CatheterTwo типа double
+	double CatheterOne, CatheterTwo;
+	// вывод вспомогательной строки в консоль
+	cout << "Length of the first catheter: "; 
+	// присваивание CatheterOne значение, введенного с консоли
+	cin >> CatheterOne;
+	// вывод вспомогательной строки в консоль
+	cout << "Length of the second catheter: ";
+	// присваивание CatheterTwo значение, введенного с консоли
+	cin >> CatheterTwo;
+
+	// вывод вспомогательной строки в консоль
+	cout << "Enter an action on the triangle (perimeter = 0, area = 1) ";
+
+	// объявление переменной input типа int
+	int input;
+	// присваивание input значение, введенного с консоли
+	cin >> input;
+
+	// объявление и инициализация константы choice
+	const auto choice = static_cast<Choice>(input);
+
+	// оператор switch
+	switch (choice)
+	{
+		case Choice::perimeter:
+        {
+		   // объявление и инициализация константы Perimeter, которой присвается значение результата вычисления функции GetPerimeter
+           const double Perimeter = GetPerimeter(CatheterOne, CatheterTwo);
+		   // вывод значения Perimeter
+           cout << "Triangle perimeter = " << Perimeter;
+           break;
+        }
+        case Choice::area:
+        {
+		   // объявление и инициализация константы Area, которой присвается значение результата вычисления функции GetArea
+           const double Area = GetArea(CatheterOne, CatheterTwo);
+		   // вывод значения Area
+           cout << "Area of the triangle = " << Area;
+           break;
+        }
+	}
+	return 0;
+}
+
+// функция, которая принимает в качестве двух аргументов значения типа double и возвращает значение типа double.
+double GetPerimeter(double CatheterOne, double CatheterTwo)
+{
+	return CatheterOne + CatheterTwo + sqrt(CatheterOne * CatheterOne + CatheterTwo * CatheterTwo);
+}
+
+// функция, которая принимает в качестве двух аргументов значения типа double и возвращает значение типа double.
+double GetArea(double CatheterOne, double CatheterTwo)
```

Comment:
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.